cannon/mipsevm/exec/mips_instructions.go

```diff
@@ -1,96 +1,114 @@
 package exec
 
 import (
+	"fmt"
+
 	"github.com/ethereum-optimism/optimism/cannon/mipsevm"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/arch"
 	"github.com/ethereum-optimism/optimism/cannon/mipsevm/memory"
+
+	// TODO: MIPS64 port. Replace with a custom library
+	u128 "lukechampine.com/uint128"
 )
 
 const (
 	OpLoadLinked       = 0x30
 	OpStoreConditional = 0x38
 )
 
-func GetInstructionDetails(pc uint32, memory *memory.Memory) (insn, opcode, fun uint32) {
+func GetInstructionDetails(pc Word, memory *memory.Memory) (insn, opcode, fun uint32) {
 	insn = memory.GetMemory(pc)
 	opcode = insn >> 26 // First 6-bits
 	fun = insn & 0x3f   // Last 6-bits
 
 	return insn, opcode, fun
 }
 
-func ExecMipsCoreStepLogic(cpu *mipsevm.CpuScalars, registers *[32]uint32, memory *memory.Memory, insn, opcode, fun uint32, memTracker MemTracker, stackTracker StackTracker) (memUpdated bool, memAddr uint32, err error) {
+func ExecMipsCoreStepLogic(cpu *mipsevm.CpuScalars, registers *[32]Word, memory *memory.Memory, insn, opcode, fun uint32, memTracker MemTracker, stackTracker StackTracker) (memUpdated bool, memAddr Word, err error) {
 	// j-type j/jal
 	if opcode == 2 || opcode == 3 {
-		linkReg := uint32(0)
+		linkReg := Word(0)
 		if opcode == 3 {
 			linkReg = 31
 		}
 		// Take top 4 bits of the next PC (its 256 MB region), and concatenate with the 26-bit offset
```

Comment:
(nit) Stale comment - should be something like?:
```suggestion
		// Take the top bits of the next PC (its 256 MB region), and concatenate with the 26-bit offset
```